### PR TITLE
Feature critical demand 

### PR DESCRIPTION
**Pull request by adnanalakori**
Just re-define constants..
Fixes #165 

By the way: once I made a PR, it shows with a new number #166 -> should be a way to keep this PR within the previous one #165?

### CODE REVIEW

**Review by adnanalakori — COMMENTED** [review]
I am not sure about this commit!

**Comment by smartie2076** [comment]
> Just re-define constants.. Fixes #165

Hi @adnanalakori! I am not sure from the title and text what you are planning in this PR, and will have to check this from the code. You should try to use a title that is different from #165, that directly tells me what you do.

Also, "fix"/"close" are github-specific words that imply an action that github will take. They are used to close issues, ie. to adress and solve an issue. Used here, I am not sure what it will do with the other PR.

> By the way: once I made a PR, it shows with a new number #166 -> should be a way to keep this PR within the previous one #165?

I think you created a new PR, because you pushed to a different branch name. The other branch is `feature_critical_demand` and not `feature_critical_demand_`. If you want to continue working on #165 you need to push onto `feature_critical_demand`!

Should I then even look at this?

File: src/constants.py [hunk 1]
```diff
@@ -40,12 +40,16 @@
...
+TITLE_DEMAND_DC = "title_demand_dc_non_critical"
+TITLE_DEMAND_DC = "title_demand_dc_critical"
 TITLE_PV = "title_pv"
 TITLE_WIND = "title_wind"
-DEMAND_AC = "demand_ac"
-DEMAND_DC = "demand_dc"
+DEMAND_AC = "demand_ac_non_critical"
+DEMAND_AC = "demand_ac_critical"
+DEMAND_DC = "demand_dc_non_critical"
+DEMAND_DC = "demand_dc_critical"
```
Comment (smartie2076) [inline src/constants.py:52]: You now have redefined `DEMAND_AC` and `DEMAND_DC` directly after you defined it. Variables have to be unique.

File: src/constants.py [hunk 2]
```diff
@@ -40,12 +40,16 @@
 PARAMETERS = "parameters"
 TARIFF = "tariff"
 TITLE_TIME = "title_time"
-TITLE_DEMAND_AC = "title_demand_ac"
-TITLE_DEMAND_DC = "title_demand_dc"
+TITLE_DEMAND_AC = "title_demand_ac_non_critical"
+TITLE_DEMAND_AC = "title_demand_ac_critical"
+TITLE_DEMAND_DC = "title_demand_dc_non_critical"
+TITLE_DEMAND_DC = "title_demand_dc_critical"
```
Comment (smartie2076) [inline src/constants.py:46]: `TITLE_DEMAND_AC` and `TITLE_DEMAND_DC` are defined twice, this does not work.

File: src/G2a_oemof_busses_and_componets.py [hunk 3]
```diff
@@ -709,7 +709,7 @@ def distribution_grid_ac(
     return distribution
 
 
-def demand_ac(micro_grid_system, bus_electricity_ac, demand_profile):
+def demand_ac(micro_grid_system, bus_electricity_ac, demand_profile, demand_name):
```
Comment (smartie2076) [inline src/G2a_oemof_busses_and_componets.py:712]: What are you planning to do with `demand_name`?

File: src/G2a_oemof_busses_and_componets.py [hunk 4]
```diff
@@ -721,10 +721,11 @@ def demand_ac(micro_grid_system, bus_electricity_ac, demand_profile):
     )
 
     micro_grid_system.add(sink_demand_ac)
+    demand_name.add(sink_demand_ac)
```
Comment (smartie2076) [inline src/G2a_oemof_busses_and_componets.py:724]: I do not understand what you use `demand_name` for, and why you use `add()`.

**Review by smartie2076 — CHANGES_REQUESTED** [review]
`cbc.exe` and `coin-license.txt` should not be pushed to the repo. I am not clear what you are planning to do...

**Comment by adnanalakori** [comment]
> Should I then even look at this?

Hi @smartie2076, sorry it just a mistake that a slash `_ `was added to the  `feature_critical_demand`, then a new PR has been opened, I was wondering of that :-) Could we delate this PR completely?

File: src/constants.py (hunk 1 above)
Comment (adnanalakori) [reply src/constants.py:52]: Ok, got it, w'll correct it -> hoppfully it runs w/o errors

File: src/G2a_oemof_busses_and_componets.py (hunk 4 above)
Comment (adnanalakori) [reply src/G2a_oemof_busses_and_componets.py:724]: I was just planning just to merge the functions `generate.demand_ac` and `generate.demand_dc` to function `generate.demand`, but not sure about the structure of this function below: 
https://github.com/rl-institut/offgridders/blob/a8fa6f1a6b8319962e65505320f9b47267fbef0a/src/G2a_oemof_busses_and_componets.py#L704

Also, just trying to reach your goal mentioned in the #162, that the new demand shall be like that : 
```
sink_demand_ac_critical = generate.demand(
       micro_grid_system, bus_electricity_ac, experiment[DEMAND_PROFILE_AC_CRITICAL], demand_name=DEMAND_AC_CRITICAL
   )
```

File: src/constants.py (hunk 2 above)
Comment (adnanalakori) [reply src/constants.py:46]: > `cbc.exe` and `coin-license.txt` should not be pushed to the repo. I am not clear what you are planning to do...

I don't know how `cbc.exe` was pushed, weird...! I just received this  error message "Error: No executable found for solver 'cbc'", then I copied it again to offgridders folder and re-run it. 

Do you know this always why this happening ?

File: src/G2a_oemof_busses_and_componets.py (hunk 4 above)
Comment (smartie2076) [reply src/G2a_oemof_busses_and_componets.py:724]: > I was just planning just to merge the functions generate.demand_ac and generate.demand_dc to function generate.demand, but not sure about the structure of this function below

Yeah, that is a good idea! You will have to change

`def demand(micro_grid_system, bus_electricity_ac, demand_profile, demand_name):`

> Also, just trying to reach your goal mentioned in the #162, that the new demand shall be like that :
> [Code]

This is correct :)

File: src/G2a_oemof_busses_and_componets.py (hunk 3 above)
Comment (smartie2076) [reply src/G2a_oemof_busses_and_componets.py:712]: Now I understand it is correct.

File: src/constants.py (hunk 2 above)
Comment (smartie2076) [reply src/constants.py:46]: Yeah, when you commit, by default all changes and added files will be added to your commit. You have to unselect the file from the upper box (the one over the commit message box):

grafik

File: src/constants.py (hunk 1 above)
Comment (smartie2076) [reply src/constants.py:52]: Will run without errors, but not nesessarily run as intended :D

**Pull request closed without merging**